docs: adapt README to crates.io

Razaloc opened the pull request:
At this moment bmap-rs and bmap-parser share the same README. This is problematic because in crates.io, once published, bmap-parser contains references to bmap-rs while if should be independent from it.

Fix the original README file.
Add a new README file for bmap-parser.

Signed-off-by: Rafael Garcia Ruiz <rafael.garcia@collabora.com>

Razaloc committed:
doc: adapt README to crates.io

bmap-rs and bmap-parser share the same README at this moment. This is
problematic because in crates.io once published bmap-parser contains
references to bmap-rs while if should be independent of it.

Fix the original README file.
Add a new README file for bmap-parser.

Signed-off-by: Rafael Garcia Ruiz <rafael.garcia@collabora.com>

Razaloc commented:
Just in case we want to quick fix the README problem, I've committed a bump to versions.

obbardc commented on bmap-parser/README.md:
```diff
@@ -0,0 +1,18 @@
+# bmap-parser
+
+The bmap-parser crate aims to implement tools related to bmap. The inspiration for it
```
bmap-parser doesn't implement tools, it implements the parsing bits for bmap files :-)

Razaloc committed:
Expand README

Explain context information for understanding bmap and bmap-rs.
Most of this information has been obtain and adapted from;
https://github.com/intel/bmap-tools/blob/master/README.md

Signed-off-by: Rafael Garcia Ruiz <rafael.garcia@collabora.com>

Razaloc committed:
Rephrase license reference

Uses a more standard reference to the license recommended by the rust
documentation (https://rust-lang.github.io/api-guidelines/necessities.html)

Signed-off-by: Rafael Garcia Ruiz <rafael.garcia@collabora.com>

obbardc commented:
can you drop the bors commit and the version bump commit ?
We are gonna drop bors at some point, and I think we should handle the release separately.

sjoerdsimons commented:
outdated, stale, closing
